### PR TITLE
Measure test coverage using pytest-cov

### LINKED ISSUE #353: Add test coverage (Codecov)
This would help immensely towards #352 The coverage and Codecov config can be taken from `aiidalab-launch` repo.

### PR DESCRIPTION
**Pull request by unkcpz**
fixes #353

### CODE REVIEW

**Comment by unkcpz** [comment]
@danielhollas I think this is enough and once it is merge the next time PR open the coverage change will be shown, correct?

**Review by danielhollas — COMMENTED** [review]
What is the .coverage file? GitHub UI claims it is binary?

**Comment by unkcpz** [comment]
It is a coverage report data file https://coverage.readthedocs.io/en/latest/cmd.html#data-file. It is generated because I run the coverage test locally. I'll remove it.

**Comment by unkcpz** [comment]
@yakutovicha badge was added, but it is still showing "unknown". I think it will show the percent report after merged to main branch?

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [![Documentation Status](https://readthedocs.org/projects/aiidalab/badge/)](https://aiidalab.readthedocs.io/)
+[![codecov](https://codecov.io/gh/aiidalab/aiidalab/branch/develop/graph/badge.svg?token=e6brvQ5r0q)](https://codecov.io/gh/aiidalab/aiidalab)
```
Comment (yakutovicha) [inline README.md:2]: Please rename the branch `main` and remove the token - it is not needed.
```suggestion
[![codecov](https://codecov.io/gh/aiidalab/aiidalab/branch/main/graph/badge.svg)](https://codecov.io/gh/aiidalab/aiidalab)
```

**Comment by yakutovicha** [comment]
> @yakutovicha badge was added, but it is still showing "unknown". I think it will show the percent report after merged to main branch?

yep. Just merge my suggestion and it should work.

**Review by yakutovicha — APPROVED** [review]
ship it 🛥️ !

**Pull request merged**